protocols/2b3642/2b3642.ot2.apiv2.py

```diff
@@ -0,0 +1,533 @@
+from opentrons import types
+import json
+import os
+import math
+
+metadata = {
+    'protocolName': '''NEBNext Ultra II DNA Library Preparation Kit
+                       for Illumina E7645S''',
+    'author': 'Sakib <sakib.hossain@opentrons.com>',
+    'description': 'Custom Protocol Request',
+    'apiLevel': '2.10'
+}
+
+
+def run(ctx):
+
+    [samples, m300_mount, p20_mount, initial_denaturation_cycles,
+     denat_anneal_cycles, final_extension_cycles] = get_values(  # noqa: F821
+        "samples", "m300_mount", "p20_mount", "initial_denaturation_cycles",
+        "denat_anneal_cycles", "final_extension_cycles")
+
+    if samples < 1 or samples > 24:
+        raise Exception('Invalid number of DNA samples (must be 1-24).')
+
+    cols = math.ceil(samples/8)
+
+    ctx.set_rail_lights(True)
+
+    # Labware
+    temp_mod = ctx.load_module('temperature module gen2', 3)
+    temp_block = temp_mod.load_labware(
+                    'opentrons_24_aluminumblock_nest_2ml_screwcap')
+    tc_mod = ctx.load_module('Thermocycler Module')
+    tc_plate = tc_mod.load_labware('nest_96_wellplate_100ul_pcr_full_skirt')
+
+    mag_mod = ctx.load_module('magnetic module gen2', 4)
+    mag_plate = mag_mod.load_labware('nest_96_wellplate_100ul_pcr_full_skirt')
+
+    tips200ul = [ctx.load_labware('opentrons_96_filtertiprack_200ul', slot)
+                 for slot in [6, 9]]
+    tips20ul = [ctx.load_labware('opentrons_96_filtertiprack_20ul', slot)
+                for slot in [5]]
+    pcr_block = ctx.load_labware(
+                    'opentrons_96_aluminumblock_generic_pcr_strip_200ul', 2)
+
+    # Pipettes
+    m300 = ctx.load_instrument('p300_multi_gen2', m300_mount,
+                               tip_racks=tips200ul)
+    p20 = ctx.load_instrument('p20_single_gen2', p20_mount, tip_racks=tips20ul)
+
+    # Wells
+    enzyme_mix = temp_block['A1']
+    rxn_buff = temp_block['B1']
+    tc_sample_wells = tc_plate.wells()[:cols*8]
```

Comment:
Good catch, `:samples` definitely works better but I believe using sample numbers that are not multiples of 8 on this kit is a bit of a waste since we are forced to use the P300 multichannel. I will discuss this further with the user during the optimization period and determine if it should be refactored to work with any sample number. Will probably have to use variable pick ups with the multichannel which can be problematic.